e2e-tests/tfm/cypress/e2e/journeys/case-amendments/facility-end-date/feature-flag-enabled/gef-deal-add-multiple-amendments-impacting-facility-end-date.spec.js

```diff
@@ -0,0 +1,224 @@
+import { format } from 'date-fns';
+import relative from '../../../../relativeURL';
+import facilityPage from '../../../../pages/facilityPage';
+import { ADMIN, BANK1_MAKER1, PIM_USER_1, T1_USER_1 } from '../../../../../../../e2e-fixtures';
+import { MOCK_FACILITY_ONE } from '../../../../../fixtures/mock-gef-facilities';
+import { MOCK_APPLICATION_AIN } from '../../../../../fixtures/mock-gef-deals';
+import { DEAL_TYPE } from '../../../../../../../gef/cypress/fixtures/constants';
+import amendmentsPage from '../../../../pages/amendments/amendmentsPage';
+import dateConstants from '../../../../../../../e2e-fixtures/dateConstants';
+
+if (Cypress.env('FF_TFM_FACILITY_END_DATE_ENABLED') === 'true') {
+  context('Amendments - GEF deal add multiple consecutive amendments impacting facility end date values', () => {
+    let dealId;
+    let dealFacilities;
+
+    const Date1 = new Date('2024-01-01');
+    const Date2 = new Date('2024-02-02');
+    const Date3 = new Date('2024-03-03');
+    const Date4 = new Date('2024-04-04');
+    const Date5 = new Date('2024-05-05');
+
+    const MOCK_GEF_FACILITY_WITH_FACILITY_END_DATE = {
+      ...MOCK_FACILITY_ONE,
+      hasBeenIssued: true,
+      isUsingFacilityEndDate: false,
+      facilityEndDate: undefined,
+      bankReviewDate: Date1,
+    };
+
+    before(() => {
+      // inserts a gef deal
+      cy.insertOneGefDeal(MOCK_APPLICATION_AIN, BANK1_MAKER1).then((insertedDeal) => {
+        dealId = insertedDeal._id;
+        // updates a gef deal so has relevant fields
+        cy.updateGefDeal(dealId, MOCK_APPLICATION_AIN, BANK1_MAKER1);
+
+        cy.createGefFacilities(dealId, [MOCK_GEF_FACILITY_WITH_FACILITY_END_DATE], BANK1_MAKER1).then((createdFacilities) => {
+          dealFacilities = createdFacilities.details;
+        });
+
+        cy.submitDeal(dealId, DEAL_TYPE.GEF, T1_USER_1);
+      });
+    });
+
+    beforeEach(() => {
+      cy.login(PIM_USER_1);
+      cy.visit(relative(`/case/${dealId}/facility/${dealFacilities._id}`));
+    });
+
+    after(() => {
+      cy.deleteDeals(dealId, ADMIN);
+    });
+
+    it('should display the current bank review date details from the facility snapshot in the Details tab', () => {
+      facilityPage.facilityIsUsingFacilityEndDate().should('have.text', 'No');
+      facilityPage.facilityBankReviewDate().should('have.text', format(Date1, 'd MMMM yyyy'));
```

Comment:
This looks to sometimes be ` 'd MMMM yyyy'` and othertimes be ` 'dd MMMM yyyy'. Perhaps making a constant at the top would ensure consistency

Reply:
Or do they genuinely alternate?

Reply:
Good idea, have done that now!